boilerplates/aws-lambda-serverless/files/esbuild-plugins.cjs

```diff
@@ -0,0 +1,58 @@
+// @ts-ignore
+const { copy } = require('esbuild-plugin-copy')
+// https://github.com/evanw/esbuild/issues/1051#issuecomment-806325487
+const nativeNodeModules = {
+  name: 'native-node-modules',
+  /**
+   * @param {{ onResolve: (arg0: { filter: RegExp; namespace: string; }, arg1: { (args: any): { path: any; namespace: string; } | undefined; (args: any): { path: any; namespace: string; }; }) => void; onLoad: (arg0: { filter: RegExp; namespace: string; }, arg1: (args: any) => { contents: string; }) => void; initialOptions: any; }} build
+   */
+  setup(build) {
+    // If a ".node" file is imported within a module in the "file" namespace, resolve
+    // it to an absolute path and put it into the "node-file" virtual namespace.
+    // @ts-ignore
+    build.onResolve({ filter: /\.node$/, namespace: 'file' }, args => {
+      // Ignore "react-dom/server.node" because it is not a native module
+      if (args.path.startsWith('react-dom/')) return undefined
+      return ({
+        // @ts-ignore
+        path: require.resolve(args.path, { paths: [args.resolveDir] }),
+        namespace: 'node-file',
+      })
+    })
+
+    // Files in the "node-file" virtual namespace call "require()" on the
+    // path from esbuild of the ".node" file in the output directory.
+    build.onLoad({ filter: /.*/, namespace: 'node-file' }, args => ({
+      contents: `
+        import path from ${JSON.stringify(args.path)}
+        try { module.exports = require(path) }
+        catch {}
+      `,
+    }))
+
+    // If a ".node" file is imported within a module in the "node-file" namespace, put
+    // it in the "file" namespace where esbuild's default loading behavior will handle
+    // it. It is already an absolute path since we resolved it to one above.
+    build.onResolve({ filter: /\.node$/, namespace: 'node-file' }, args => ({
+      path: args.path,
+      namespace: 'file',
+    }))
+
+    // Tell esbuild's default loading behavior to use the "file" loader for
+    // these ".node" files.
+    let opts = build.initialOptions
+    opts.loader = opts.loader || {}
+    opts.loader['.node'] = 'file'
+  }
+}
+
+const copyPlugin = copy({
+  assets: [
+    {
+      from: ['./dist/**/*'],
+      to: ['dist']
+    }
+  ]
+})
+
+module.exports = [nativeNodeModules /*, copyPlugin*/]
```

Comment:
Why is `copyPlugin` commented?

Reply:
It was part of the original [example setup from Strydom](https://github.com/Strydom/vite-production-serverless-build). 

My setup differs as it serves part of the static content from a s3 bucket with the route "/assets" defined in the reverse proxy (Cloudfront) setup. On the downside currently vike `prerendering=true` does not work :-(.

If I can get this working, I will need this plugin to copy the static html files to the final output which gets packaged to be deployed in the lambda. The restrictions of Cloudfront allow only 25 root rules. The current setup will handle all static assets which are not in the `client/assets` folder with the same lambda which renders the dynamic pages.

Reply:
I have added a section to the [README](https://github.com/vikejs/bati/pull/338/files#diff-2fc3a2e300f8d5aab58b3ec3ac22cd6462eb179472ce0cb59c5531ceb086c4ca) to explain the usage of the copyPlugin.